lib/web/apiserver.go

```diff
@@ -2659,63 +2659,143 @@ func (h *Handler) siteNodeConnect(
 }
 
 func (h *Handler) generateSession(ctx context.Context, clt auth.ClientI, req *TerminalRequest, clusterName string, scx *SessionContext) (session.Session, error) {
-	var (
-		id   string
-		host string
-		port int
-	)
 	owner := scx.cfg.User
 	h.log.Infof("Generating new session for %s\n", clusterName)
 
-	if _, err := uuid.Parse(req.Server); err != nil {
+	host, err := findByHost(ctx, clt, req.Server)
```

Comment:
cc @rosstimothy - is this going to negate some of our web UI connection latency improvements?